components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabViewHolder.kt

```diff
@@ -47,7 +49,13 @@ class TabViewHolder(
         }
 
         titleView.text = title
-        urlView?.text = tab.url
+
+        var url = tab.url
+        try {
+            url = URL(tab.url).host
+        }
+        catch (e: MalformedURLException) { /* ignore error */ }
+        urlView?.text = url
```

Comment:
Let's use the catch clause properly to set the fallback URL if the parsing fails.

```suggestion
        try {
            val tabHost = URL(tab.url).host
            urlView?.text = tabHost
        } catch (e: MalformedURLException) {
            urlView?.text = tab.url
        }
```

It's generally not good practice to not handle the exceptions. If this is not the place you want to handle it, you can annotate the as a method that throws exceptions and the caller will then have to put the try-catch clause appropriately

Reply:
nit: we should lift the assignment out of the try catch:

```kotlin
urlView?.text = try {
    URL(tab.url).host          
} catch (e: MalformedURLException) {
    tab.url
}
```

Reply:
We also have an extension function for this we could re-use here: https://github.com/mozilla-mobile/android-components/blob/f362c94e32c715f334b3880f7bb696a94cdbca6a/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt#L107

So, you can just call:

```kotlin
tab.url.tryGetHostFromUrl()
```